frame/dapps-staking/src/tests.rs

```diff
@@ -1708,6 +1708,203 @@ fn claim_is_ok() {
+        // reward destination to DelegateBalance
+        assert_noop!(
+            DappsStaking::claim_staker(
+                RuntimeOrigin::signed(first_staker),
+                first_contract_id,
+            ),
+            Error::<TestRuntime>::DelegateeAccountNotSetForContract
+        );
+
+        add_delegatee_address(first_staker, first_contract_id.clone(), first_delegatee_address).unwrap();
+        add_delegatee_address(first_staker, second_contract_id.clone(), second_delegatee_address).unwrap();
+
+        assert_claim_staker(first_staker,&first_contract_id);
+        assert_claim_staker(first_staker,&second_contract_id);
+
+        // Now changing reward Destination should remove all the delegatee mapping
+        assert_set_reward_destination(first_staker, RewardDestination::FreeBalance);
+
+        assert!(DelegateInfo::<TestRuntime>::get(&first_staker,&first_contract_id).is_none(),"Mapping Removed");
+        assert!(DelegateInfo::<TestRuntime>::get(&first_staker,&second_contract_id).is_none(),"Mapping Removed");
+
+    })
+    
+}
+
+#[test]
+fn set_delegatee_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let first_developer = 1;
+        
+        let first_contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        
+        let alice =1;
+        let bob = 2;
+        
+        // Contract must be registered is order to set delegatee
+        assert_noop!(DappsStaking::set_delegatee(
+            RuntimeOrigin::signed(alice),
+            first_contract_id,
+            bob
+        ),Error::<TestRuntime>::NotOperatedContract);
+        
+        assert_register(first_developer, &first_contract_id);
+
+        assert!(DappsStaking::set_delegatee(
+            RuntimeOrigin::signed(alice),
+            first_contract_id,
+            bob
+        ).is_ok());
+
+        assert_eq!(DelegateInfo::<TestRuntime>::get(&alice,&first_contract_id).unwrap(),bob);
+
+    })
+}
+
+#[test]
+fn remove_delegatee_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let first_developer = 1;
+        
+        let first_contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        
+        let alice =1;
+        let bob = 2;
+        
+        //check can't remove until registered
+
+        assert_noop!(DappsStaking::remove_delegatee(
+            RuntimeOrigin::signed(alice),
+            first_contract_id
+        ),
+        Error::<TestRuntime>::DelegateeAccountNotSetForContract);
+
+        assert_register(first_developer, &first_contract_id);
+
+        assert!(DappsStaking::set_delegatee(
+            RuntimeOrigin::signed(alice),
+            first_contract_id,
+            bob
+        ).is_ok());
+
+        assert!(DappsStaking::remove_delegatee(
+            RuntimeOrigin::signed(alice),
+            first_contract_id
+        ).is_ok());
```

Comment:
It would be good to have these actions checked in the same way that all other extrinsic calls are checked.

But I also like the fact that you added unit tests :slightly_smiling_face: